Use tags to reduce project update output

AlanCoding opened the pull request:
##### SUMMARY
A major concern of mine is that the project update output is hard for humans to digest.

I would like to do something that both aligns with our future direction, and makes the project update output a little easier on the eyes. This solution isn't perfect, but there's a good case to be made that it's _better_.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```
9.0.0
```


##### ADDITIONAL INFORMATION
I have two scenarios to present here. The first is a "normal" project update done by clicking the 🚀 button manually.

![Screen Shot 2019-11-01 at 3 05 36 PM](https://user-images.githubusercontent.com/1385596/68049661-d8ebeb80-fcb9-11e9-818d-74acac837dc1.png)

The `JOB TAGS` is our means of reporting what is or isn't ran. In this case, we do a git update. Not rocket science. Everyone with me?

Here's the project sync in advance of a job that has collection requirements.

![Screen Shot 2019-11-01 at 3 04 42 PM](https://user-images.githubusercontent.com/1385596/68049732-0042b880-fcba-11e9-8368-d7ebc3ff5099.png)

(I think I need UI changes here to show the tags, they should be there)

This _particular_ project sync did _not_ need to update the source tree. However, it _did_ need to pull in new collection requirements, so that's what changed in this case.

The goal is that _at a glance_, the user can see what was _done_ and what was _changed_, and that the different sources of information on the page are relatively self-consistent and self-explanatory.

AlanCoding committed:
Use tags to reduce project update output

Handle folder deletion as tag

remove -v use by default

Change meaning of roles_enabled playbook var to
 value of AWX global setting

wenottingham commented on awx/playbooks/project_update.yml:
```diff
@@ -63,8 +64,9 @@
-      when: scm_type == 'hg'
       delegate_to: localhost
+      tags:
+        - checkout_hg
```
update_&lt;scm&gt;?

jakemcdermott commented:
_What if_ we removed everything including and beyond the `=>` from the displayed output for tasks - In general and for all jobs. 

The mockups even show it this way: https://tower-mockups.testing.ansible.com/mockups/jobs/jobs-details/ (note the lack of `=> {...` spam next to the task lines)

Were we to put the abbreviated task stdout in a new API field and display it in the UI, we could still show the fact data in the task details modal so that it's accessible if someone needs it. This would result in much cleaner looking output that's easier for people to read _at a glance_, assuming the playbook author made the effort to write good names for their tasks.

jakemcdermott commented:
Given that some of the more important bits (stdout for the task's module, etc.) are already available in the task details modal, how useful is this information at a glance, _really_ ?

![Screenshot from 2019-11-01 18-05-24](https://user-images.githubusercontent.com/9753817/68059298-57ed1e00-fcd2-11e9-873e-b04b5e1a62c2.png)

Instead, this could be:

```
TASK [detect collections/requirements.yml] ******************************** 15:03:51
ok: [localhost -> localhost]

TASK [fetch galaxy collections from collections/requirements.yml] ********* 15:03:52
changed: [localhost -> localhost]
```

...With the option to see all the additional `=> {...` information (module stdout, stderr, facts, etc.) in the task modal.

As I mentioned above, separating the abbreviated task output from the rest of the stdout and putting the results into their own fields is a requirement for this, since the UI really can't be in the business of splitting, parsing, and categorizing ansible's raw text data dump for you.

AlanCoding commented:
I agree with all of the comments by @wenottingham and @jakemcdermott and have implemented them. I also discovered that the tags not showing up was already a resolved UI issue and I just had not refreshed my UI. With those all resolved, updated screenshots:

![Screen Shot 2019-11-02 at 10 56 39 PM](https://user-images.githubusercontent.com/1385596/68079938-6a498380-fdc8-11e9-9b95-233b94f3e369.png)

With delete on update selected:

![Screen Shot 2019-11-02 at 11 24 28 PM](https://user-images.githubusercontent.com/1385596/68079940-77667280-fdc8-11e9-88a7-162530dc1790.png)

The comparatively more complex case of inventory update w/ collection:

![Screen Shot 2019-11-02 at 11 21 38 PM](https://user-images.githubusercontent.com/1385596/68079944-88af7f00-fdc8-11e9-83e7-b0ae9d9fe92b.png)

Because I removed the `-v` which was used as the default verbosity, I had added back in the task that is a simple debug message with the repository version. This _is_ redundant with other information on the page, but the playbook context and the details pane involve a non-trivial data exchange. So this is more about painting a story about where the revision came from and where it went.

AlanCoding commented:
Looked over all the testing, and I think this is in a mergeable state now. @matburt @trahman73, any thoughts on the look and feel with this proposed change?

trahman73 commented:
@AlanCoding Design looks good to me.

ryanpetrello commented on awx/main/migrations/0099_v370_projectupdate_job_tags.py:
```diff
@@ -0,0 +1,18 @@
+        migrations.AddField(
+            model_name='projectupdate',
+            name='job_tags',
+            field=models.CharField(blank=True, default='', help_text='Parts of the project update playbook that will be ran.', max_length=1024),
```
That will be _run_ ?
